Automattic/wp-calypso
# Stats: Add Jetpack Stats products supporting check

Related to #78654 

## Proposed Changes

The PR adds the ability to check whether a site has Stats product or a plan that supports Stats.

#### Selectors:
- `hasSiteProductJetpackStats`: returns true if the site has has Paid or Free products
- `hasSiteProductJetpackStatsPaid`: returns true if the site has Paid Stats products

The PR also introduces `FEATURE_JETPACK_STATS_PAID` and `FEATURE_JETPACK_STATS_FREE` as features, which would be used for plans that support Jetpack Stats Paid or Free.


## Testing Instructions

* Sandbox Store following instructions at pejTkB-FS-p2
* Purchase Stats for a site `https://wordpress.com/checkout/{site-slug}/jetpack_stats_monthly`
* Open Calypso Live
* Add a couple of lines somewhere, for example: https://github.com/Automattic/wp-calypso/blob/aa7dfcbfe0d3eb0ef5c6e2020bfd9fba68bacfff/client/my-sites/stats/site.jsx#L475

```JavaScript
		const hasJetpackStats = hasSiteProductJetpackStats( state );
		const hasJetpackStatsPaid = hasSiteProductJetpackStatsPaid( state );
		console.log( 'hasJetpackStats', hasJetpackStats );
		console.log( 'hasJetpackStatsPaid', hasJetpackStatsPaid );
```

<img width="700" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/bae015c2-2894-4d92-a4be-35a49fc9ad1e">

- Ensure the output in console is expected for sites


## Pre-merge Checklist
